spec/models/mixins/miq_policy_mixin_spec.rb

```diff
@@ -0,0 +1,72 @@
+describe MiqPolicyMixin do
+  let(:host) { FactoryBot.create(:host) }
+  let(:policy) { FactoryBot.create(:miq_policy) }
+  let(:policy_set) { FactoryBot.create(:miq_policy_set).tap { |ps| ps.add_member(policy) } }
+
+  describe "#get_policies" do
+    it "supports no policies" do
+      expect(host.get_policies).to eq([])
+    end
+
+    it "supports policies" do
+      host.add_policy(policy)
+      expect(host.get_policies).to eq([policy])
+    end
+
+    it "supports policy sets" do
+      host.add_policy(policy_set)
+      expect(host.get_policies).to eq([policy_set])
+    end
+  end
+
+  describe "#has_policies?" do
+    it "detects no policies" do
+      expect(host.has_policies?).to eq(false)
+    end
+
+    it "detects policies" do
+      host.add_policy(policy)
+      expect(host.has_policies?).to eq(true)
+    end
+
+    it "detects policy sets" do
+      host.add_policy(policy_set)
+      expect(host.has_policies?).to eq(true)
+    end
+  end
+
+  describe "#has_policies" do
```

Comment:
I expected `has_policies` and `has_policies?` tests to be identical, since they are aliases, but they aren't which I don't understand why?

Reply:
I'll make this more clear. one is testing the SQL attribute, the other is testing the ruby attribute.

Reply:
is this more clear now?